ofdb-db-sqlite/src/repo_impl/place.rs

```diff
@@ -684,3 +704,61 @@ fn load_place_revision(
         .map_err(from_diesel_err)?;
     load_place(conn, row)
 }
+
+const EXCLUDE_STATUS: &[ReviewStatus] = &[ReviewStatus::Archived, ReviewStatus::Rejected];
+
+fn find_places_not_updated_since(
+    conn: &mut SqliteConnection,
+    not_updated_since: Timestamp,
+) -> Result<Vec<(Place, ReviewStatus)>> {
+    use schema::{place::dsl, place_revision::dsl as rev_dsl};
+
+    let mut query = schema::place_revision::table
+        .inner_join(
+            schema::place::table.on(rev_dsl::parent_rowid
+                .eq(dsl::rowid)
+                .and(rev_dsl::rev.eq(dsl::current_rev))),
+        )
+        .select((
+            rev_dsl::rowid,
+            rev_dsl::rev,
+            rev_dsl::created_at,
+            rev_dsl::created_by,
+            rev_dsl::current_status,
+            rev_dsl::title,
+            rev_dsl::description,
+            rev_dsl::lat,
+            rev_dsl::lon,
+            rev_dsl::street,
+            rev_dsl::zip,
+            rev_dsl::city,
+            rev_dsl::country,
+            rev_dsl::state,
+            rev_dsl::contact_name,
+            rev_dsl::email,
+            rev_dsl::phone,
+            rev_dsl::homepage,
+            rev_dsl::opening_hours,
+            rev_dsl::founded_on,
+            rev_dsl::image_url,
+            rev_dsl::image_link_url,
+            dsl::id,
+            dsl::license,
+        ))
+        .order_by(rev_dsl::created_at.desc())
+        .filter(rev_dsl::created_at.lt(not_updated_since.as_millis()))
+        .into_boxed();
+
+    for status in EXCLUDE_STATUS {
+        query = query.filter(rev_dsl::current_status.ne(ReviewStatusPrimitive::from(*status)));
+    }
+
+    let rows = query
+        .load::<models::JoinedPlaceRevision>(conn)
+        .map_err(from_diesel_err)?;
+    let mut results = Vec::with_capacity(rows.len());
```

Comment:
Collecting results and aborting on error should work.

> ...and an iterator of [Result<T, E>](https://doc.rust-lang.org/std/result/enum.Result.html) items can be collected into Result<Collection<T>, E>.

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.collect

Reply:
Example: https://github.com/aoide-org/aoide-rs/commit/6bd71cb106403d78f1307321749710f1c11a3afd